src/commands/account/delete_key/mod.rs

```diff
@@ -26,23 +28,36 @@ impl DeleteKeyCommandContext {
         Ok(Self {
             config: previous_context.0,
             owner_account_id: scope.owner_account_id.clone().into(),
-            public_key: scope.public_key.clone().into(),
+            public_keys: scope
+                .public_keys
+                .clone()
+                .0
+                .into_iter()
+                .map(Into::into)
+                .collect(),
         })
     }
 }
 
 impl From<DeleteKeyCommandContext> for crate::commands::ActionContext {
     fn from(item: DeleteKeyCommandContext) -> Self {
+        let public_keys_clone = std::sync::Arc::new(item.public_keys.clone());
         let on_after_getting_network_callback: crate::commands::OnAfterGettingNetworkCallback =
             std::sync::Arc::new(move |_network_config| {
                 Ok(crate::commands::PrepopulatedTransaction {
                     signer_id: item.owner_account_id.clone(),
                     receiver_id: item.owner_account_id.clone(),
-                    actions: vec![near_primitives::transaction::Action::DeleteKey(
-                        near_primitives::transaction::DeleteKeyAction {
-                            public_key: item.public_key.clone(),
-                        },
-                    )],
+                    actions: public_keys_clone
+                        .clone()
+                        .iter()
+                        .map(|public_key| {
+                            near_primitives::transaction::Action::DeleteKey(
+                                near_primitives::transaction::DeleteKeyAction {
+                                    public_key: public_key.clone(),
```

Comment:
I believe this should work and void unnecessary cloning:

```suggestion
                        .into_iter()
                        .map(|public_key| {
                            near_primitives::transaction::Action::DeleteKey(
                                near_primitives::transaction::DeleteKeyAction {
                                    public_key,
```